### ISSUE TITLE
Error: Non-space character inside "noscript" inside "head", due to HTML minification

### ISSUE THREAD

**Issue by andersk** [farm-fe/farm#1210]
### Steps to reproduce

In an empty directory, create `index.html`:

```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Farm</title>
  </head>
  <body>
    <noscript>This app requires JavaScript.</noscript>
  </body>
</html>
```

and run `farm build`.

### What is expected?

The build should succeed.

### What is actually happening?

Farm deletes the `<body>` tag as part of HTML minification, and then complains that the following `<noscript>` tag is invalid (which it wasn’t before, but now is).

```console
$ npx -y -p @farmfe/core -p @farmfe/cli farm build
[ building ] ⠂ transform (18) ../../../.npm/_npx/947b0871b6ccd760/node_modules/@swc/helpers/esm/_export_star.js.farm-runtime
[ Farm ] Error: Parse `index_0eb5_html` failed.
 Error: 
  × Non-space character inside "noscript" inside "head"
   ╭─[index_0eb5_html:1:1]
 1 │ <!doctype html><meta charset=utf-8><title>Farm</title><noscript>This app requires JavaScript.</noscript>
   ·                                                                 ─
   ╰────

  × Stray end tag "noscript"
   ╭─[index_0eb5_html:1:1]
 1 │ <!doctype html><meta charset=utf-8><title>Farm</title><noscript>This app requires JavaScript.</noscript>
   ·                                                                                              ───────────
   ╰────

Potential Causes:
1.The module have syntax error.
2.This kind of module is not supported, you may need plugins to support it
```

### System Info

```
  System:
    OS: Linux 6.8 NixOS 24.05 (Uakari) 24.05 (Uakari)
    CPU: (12) x64 Intel(R) Core(TM) i7-10710U CPU @ 1.10GHz
    Memory: 8.38 GB / 15.26 GB
    Container: Yes
    Shell: 5.2.26 - /run/current-system/sw/bin/bash
  Binaries:
    Node: 20.11.1 - ~/.nix-profile/bin/node
    Yarn: 1.22.19 - ~/.nix-profile/bin/yarn
    npm: 10.2.4 - ~/.nix-profile/bin/npm
    pnpm: 9.0.3 - ~/.nix-profile/bin/pnpm
    bun: 1.1.1 - ~/bin/bun
  Browsers:
    Chromium: 123.0.6312.122
```

**Comment by andersk** [comment]
I can’t seem to reproduce this with direct use of `swc_html_minifier`, so I suspect the bug must be in Farm itself, unless there’s some configuration option I’m missing.

```rust
use swc_common::{BytePos, FileName, SourceFile};
use swc_html::codegen::{writer::basic::BasicHtmlWriter, CodeGenerator, Emit};
use swc_html::parser::parse_file_as_document;
use swc_html_minifier::minify_document;

fn main() {
    let source_file = SourceFile::new(
        FileName::Anon,
        false,
        FileName::Anon,
        "<!doctype html>
<html>
  <head>
    <meta charset=\"utf-8\" />
    <title>Farm</title>
  </head>
  <body>
    <noscript>This app requires JavaScript.</noscript>
  </body>
</html>"
            .to_string(),
        BytePos(1),
    );
    let mut errors = vec![];
    let mut document =
        parse_file_as_document(&source_file, Default::default(), &mut errors).unwrap();
    minify_document(&mut document, &Default::default());
    let mut out = String::new();
    let mut code_generator = CodeGenerator::new(
        BasicHtmlWriter::new(&mut out, None, Default::default()),
        Default::default(),
    );
    code_generator.emit(&document).unwrap();
    println!("{}", out);
}
```

```html
<!DOCTYPE html>
<html>
<head><meta charset="utf-8" /><title>Farm</title></head><body><noscript>This app requires JavaScript.</noscript></body></html>
```

**Comment by andersk** [comment]
`farm build` succeeds with `minify: false` in `farm.config.js`.

```js
import { defineConfig } from "@farmfe/core";
export default defineConfig({
  compilation: {
    minify: false,
  },
});
```

**Comment by andersk** [comment]
Oh I see, I needed `CodegenConfig { minify: true, … }`. I’ll file an SWC issue then.

**Comment by wre232114** [comment]
Thanks for reporting issues, I'll look into it later

**Comment by andersk** [comment]
- Opened swc-project/swc#8876.

**Comment by andersk** [comment]
My fix was merged in swc_html_codegen 0.42.27.

**Comment by wre232114** [comment]
I'll upgrade swc in `@farmfe/core@1.1.0` today
